### ISSUE TITLE
Trial minimal reporter prints a negative run time in summary

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3184]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#3184|
|Type|defect|
|Created|2008-04-16 18:16:28Z|
|Branch|https://github.com/twisted/twisted/tree/summary-reporter-3184|

For example, on my machine:
```
$ trial --reporter=summary twisted.test.test_usage

-------------------------------------------------------------------------------
Ran 22 tests in 0.053s

-0.0538351535797 22 22 0 0 0
```

<details><summary>Searchable metadata</summary>

```
trac-id__3184 3184
type__defect defect
reporter__therve therve
priority__normal normal
milestone__ 
branch__branches_summary_reporter_3184 branches/summary-reporter-3184
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1208369788000000 1208369788000000
changetime__1212413820000000 1212413820000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [23790]) Branching to 'summary-reporter-3184'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [23791]) Fix and test.

Refs [#3184](https://github.com/twisted/twisted/issues/3184)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve commented|
|-|-|

I also fixed a problem when running a test file without tests.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @therve<br>@exarkun set status to `new`|
|-|-|

An extra word in the docstring for `test_emptyMinimalReporter` - _zeroes if when no test_.

It'd be super awesome to know what all those numbers are.  I'll settle for an addition to the docstring of `test_minimalReporter` which explains that the first element is the duration of the test run in seconds, though.

Another great addition would be API docs for `_startTime` in the class docstring for `Reporter`.

There seems to be a functional issue with the test as well, unfortunately.  Windows has a pretty low resolution clock, I think the time can be 0 sometimes.  A builder had a failure:

```
[FAIL]: twisted.trial.test.test_reporter.PyunitTestNames.test_minimalReporter

Traceback (most recent call last):
  File "s:\buildbots\twisted\server-2k8-x86-py2.5-select\Twisted\twisted\trial\test\test_reporter.py", line 332, in test_minimalReporter
    self.assertTrue(timer > 0)
twisted.trial.unittest.FailTest: None
```

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve commented|
|-|-|

(In [23808]) Process review comments

Refs [#3184](https://github.com/twisted/twisted/issues/3184)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve removed owner|
|-|-|

It should be better. It highlighted some pretty unfortunate things in MinimalReporter though.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @therve|
|-|-|

`test_emptyMinimalReporter` seems to be short a word now. :P

Also, please update copyright dates and give `test_reporter.py` a module docstring if you feel like it.  Then, merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_10"></a>|@therve set status to `closed`|
|-|-|

(In [23819]) Merge summary-reporter-3184

Author: therve
Reviewer: exarkun
Fixes [#3184](https://github.com/twisted/twisted/issues/3184)

Fix trial summary reporter: it now prints positive runtime and doesn't break
when there is not test.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
